app/models/reducers/summary_statistics_reducer.rb

```diff
@@ -0,0 +1,175 @@
+module Reducers
+  class SummaryStatisticsReducer < Reducer
+    attr_reader :extracts
+
+    @@valid_operations = [
+      "count",
+      "min",
+      "max",
+      "sum",
+      "product",
+      "mean",
+      "sse",
+      "variance",
+      "stdev",
+      "first",
+      "median",
+      "mode"
+    ]
+
+    validate do
+      unless config["summarize_field"].present?
+        errors.add("summarize_field", "No field specified for operations")
+      else
+        if config["summarize_field"].ends_with? "."
+          errors.add("summarize_field", "Invalid summarize_field specified")
+        end
+      end
+
+      unless config["operations"].present?
```

Comment:
Config field needs to be defined via DSL (see other reducers) so that the web UI will know how to render the edit form.

Reply:
done, but is there a way to use the DSL to indicate that i'm looking to accept either a single string OR an array of strings here?

Reply:
No. Can we just require an array with one string in that case?

Reply:
i thought it might be nice not to have to create three or four copies of this reducer if you wanted to calculate like, the standard deviation AND the mean of a given field.

Reply:
the `fieldmap` for a `PluckFieldExtractor` has a structure as well so we might have to solve that too

Reply:
I'm fine with accepting an array of strings. Just wondering if we need to allow all of:

```
"mean"
["mean"]
["mean", "variance"]
```

My though was to drop the first one, and just require people to use the second form if they only want one operation.

Reply:
oh sure, that's fine! we already do that for the extract filter on reducers too, it's not a big deal. is there a way to express that in the DSL?